src/lib/guesserStrategies/retryMisplacedLettersStrategy.ts

```diff
@@ -0,0 +1,22 @@
+import { IStrategyScoreMethod, NextWordGuesserStrategyBase } from '../nextWordGuesserStrategy';
+import { LetterAtPositionInWord, LetterWithPosition } from '../letterAtPosition';
+
+export default class RetryMisplacedLettersStrategy extends NextWordGuesserStrategyBase implements IStrategyScoreMethod {
+    constructor(...params: ConstructorParameters<typeof NextWordGuesserStrategyBase>) {
+        super(...params);
+    }
+
+    scoreForGuess(guess: string): number {
+        const previouslyMisplacedLetters = this.wordList.letterRules
+            .filter((rule) => rule.required === LetterAtPositionInWord.Misplaced)
+            .map(({ letter, position }) => ({ letter, position }) as LetterWithPosition);
+        return Array.from(guess)
+            .map((letter, position) => ({ letter, position }))
+            .filter(({ letter, position }) =>
+                previouslyMisplacedLetters.some(
+                    (previouslyMisplaced) =>
+                        previouslyMisplaced.letter === letter && previouslyMisplaced.position !== position
```

Comment:
Perhaps simplify this `.filter(... => some(...))` usage to `intersectionBy` or some such?

Reply:
Code cleanup for readability & to be more self-documenting will be done as part of https://github.com/codergeek42/wordle-solver/issues/53 in order not to blow up scope of this PR.